jira/client.py

```diff
@@ -2927,6 +2927,19 @@ def statuses(self) -> List[Status]:
         ]
         return statuses
 
+    def project_statuses_by_issue_type(self, projectIdOrKey: str) -> List[IssueType]:
+        """Get a list of issue types available within the project with available statuses within each issue type, as each project has a set of valid issue types and each issue type has a set of valid statuses.
```

Comment:
Hi @Kayx23 !

Sorry I'm just reviewing this again and think this function name and docstring could be clarified.

Technically this function is returning issue types rather than project statuses.

So perhaps the function name should be `issue_types_for_project()`

I also think the short summary for the function could be made clearer with an example.

Perhaps we can modify this like so:

```suggestion
    def issue_types_for_project(self, projectIdOrKey: str) -> List[IssueType]:
        """Get a list of issue types available within the project.
        
         Each project has a set of valid issue types and each issue type has a set of valid statuses.
         The valid statuses for a given issue type can be extracted via: `issue_type_x.statuses`
```

We can then probably also update the docstring of `statuses()` to point to this function to help people find this use case!

```python
        """Get a list of all status Resources from the server.

           Refer to :py:meth:`JIRA.issue_types_for_project` for getting statuses
           for a specific issue type within a specific project.
```

Reply:
Good suggestion! This makes a lot of sense. I've added them in.